### PR TITLE
Editor currently executing line #20

### PR DESCRIPTION
**Pull request by michaelschuff**
Added Line Highlighting when running code woot woot.

### CODE REVIEW

**Review by TrevorBrunette — CHANGES_REQUESTED** [review]
Please do not commit the `target` folder. It is now in the git ignore if you pull from main into this branch.

**Review by TrevorBrunette — CHANGES_REQUESTED** [review]
In the function removeHighlights(JTextComponent), you assign variables "hilite" and "hilites". I think these are meant to be "highlighter" and "highlights"?

**Comment by michaelschuff** [comment]
Ight committed the changes

**Review by TrevorBrunette — CHANGES_REQUESTED** [review]
Please remove the `target` folder from the commit -- perhaps using `git rm -r --cached target`.  Parsing every line up to the current before each execution seems too inefficient, perhaps a data structure mapping `Line` numbers to the actual line of code they are on in the text could be used? If possible it would be good to make something that allows the highlight() function to seek directly to the necessary line each time.

**Comment by michaelschuff** [comment]
K I believe I merged it correctly, should be done now.

**Comment by TrevorBrunette** [comment]
The major changes to the Simulator implementation require a resolution of merge conflicts. Please merge the changes and then I will review.

**Review by TrevorBrunette — APPROVED** [review]
Looks good to me. I added some suggested changes. Please consult me before merging this. When this does get merged, please merge as a squash and merge.

**Review by TrevorBrunette — CHANGES_REQUESTED** [review]
Sorry, I did forget to mention something -- can you have line highlight follow the theme? Like on dark themes the yellow background on white text is impossible to read, so having it use a theme color for the highlighting wold make sense.

**Comment by michaelschuff** [comment]
Yeah totally. Its on the todo list right after merging it with the new sim

File: src/main/java/com/ezasm/gui/Window.java [hunk 1]
```diff
@@ -15,6 +16,13 @@
  * relevant simulator.
  */
 public class Window {
+    public class School {
```
Comment (WyattRoss) [inline src/main/java/com/ezasm/gui/Window.java]: What is this and why does it exist? My LSP finds no references to this anywhere in the project.

**Review by WyattRoss — CHANGES_REQUESTED** [review]
Appears to crash the simulation thread when a function is called. Try running function.ez to replicate.

File: src/main/java/com/ezasm/gui/Window.java (hunk 1 above)
Comment (michaelschuff) [reply src/main/java/com/ezasm/gui/Window.java]: got rid of it, ik how it got there and its quite embarrassing and id prefer not to say how nor why.

File: src/main/java/com/ezasm/gui/EditorPane.java [hunk 2]
```diff
@@ -184,6 +184,6 @@ public void updateHighlight(int line) {
     }
 
     public void resetHighlighter() {
-        highlighter = new LineHighlighter(HIGHLIGHT_COLOR, textArea);
+        highlighter = new LineHighlighter(Window.currentTheme().getCurrentLine(), textArea);
```
Comment (WyattRoss) [inline src/main/java/com/ezasm/gui/EditorPane.java:187]: getCurrentLine() is for the color of the line currently edited to be highlighted. You were probably on the right track with using some sort of yellow to be honest. My bad, getCurrentTheme is an unintuitive method name and should have had some javadoc attached to it.

Comment (michaelschuff) [reply src/main/java/com/ezasm/gui/EditorPane.java:187]: ahh got it

Comment (michaelschuff) [reply src/main/java/com/ezasm/gui/EditorPane.java:187]: Should I add a color for line highlighting? I dont see one currently

**Comment by michaelschuff** [comment]
Why is it failing the last two checks?

File: src/main/java/com/ezasm/gui/EditorPane.java (hunk 2 above)
Comment (WyattRoss) [reply src/main/java/com/ezasm/gui/EditorPane.java]: You could just use the yellow provided.

**Pull request closed without merging**